src/server/mod.rs

```diff
@@ -678,7 +387,7 @@ impl Seabird for Arc<Server> {
                     }
                     Either::Right((_, _)) => {
                         // Stream was closed by the client - this is not actually an error.
-                        Ok(())
+                        return Ok(());
```

Comment:
This was another potential cause for the panic - we didn't properly exit the function here, we just continued to the next loop iteration.